pkg/scale/encode.go

```diff
@@ -272,10 +366,14 @@ func (es *encodeState) encodeBool(l bool) (err error) {
 func (es *encodeState) encodeBytes(b []byte) (err error) {
 	err = es.encodeLength(len(b))
 	if err != nil {
+		err = fmt.Errorf("encoding []byte: %w", err)
```

Comment:
note: here we don't really want to add the full byte slice to the error context since it might be huge

```suggestion
		err = fmt.Errorf("encoding length: %w", err)
```